stage/stage-qc.R

```diff
@@ -1,19 +1,55 @@
+library(hakaiApi)
+library(readr)
+library(tidyr)
+library(dplyr)
+library(zoo)
+library(ggplot2)
+library(lubridate)
+library(glue)
+library(stringr)
+
+create_qc_table_name <- function(table_name) {
+  table_name <- tolower(table_name)
+  gsub("us:", "_", table_name)
+}
+
+qc_author <- 'emily.haughton@hakai.org'
+baseurl <- "https://goose.hakai.org/api"
+station_name <- "SSN819US"
+sampling_interval <- "5minute"
+table_name <- glue("{station_name}:{sampling_interval}")
+qc_table_name <- create_qc_table_name(table_name)
+meas_start_date <- "2016-04-01"
+meas_end_date <- "2016-04-04"
+
+# Initialize the client
+hakai_client <- hakaiApi::Client$new(baseurl)
+
 # Step 1: Load the data
-df <- read_csv(
-  "844stage.csv",
-  col_types = cols(
-    timestamp = col_datetime(format = "%Y-%m-%d %H:%M"),
-    qlevel = col_number(),
-    qflag = col_character(),
-    stage = col_number()
-  )
-)
+## contruct the url
+url <- glue("{baseurl}/sn/views/{table_name}Samples?measurementTime>={meas_start_date}&measurementTime<{meas_end_date}&limit=-1")
+df_raw <- hakai_client$get(url)
+
+## set types explicitly
+df <- df_raw |> 
+  mutate(across(ends_with("_QC"), as.character)) |> 
+  mutate(across(!ends_with("_QC") & !any_of("measurementTime"), as.double)) |> 
+  mutate(measurementTime = as_datetime(measurementTime))
+
+## generic column renaming
+df <- df %>%
+  rename_with(~ case_when(
+    str_ends(.x, ":PLS2_Lvl$") ~ "stage",
+    str_ends(.x, ":PLS2_Lvl_QL$") ~ "qlevel", 
+    str_ends(.x, ":PLS2_Lvl_QC$") ~ "qflag",
+    TRUE ~ .x
+  ))
 
 # Step 2: Apply linear interpolation for small gaps (<12)
 df <- df %>%
-  arrange(timestamp) %>%
+  arrange(measurementTime) %>%
   mutate(
-    stage_filled = zoo::na.approx(stage, x = timestamp, maxgap = 12, na.rm = FALSE),
+    stage_filled = zoo::na.approx(stage, x = measurementTime, maxgap = 12, na.rm = FALSE),
     qflag = case_when(
       is.na(stage) & !is.na(stage_filled) ~ "EV_LINEAR",  # Linear fill
       TRUE ~ qflag
```

Comment:
you might need to cast qflag to a character vector in cases where qflag is already set as R seems to think it's a number column. `TRUE ~ as.character(qflag)`

Reply:
fixed in https://github.com/HakaiInstitute/wx-tools/pull/2/commits/d04d72a89c97ef68fbe788e792adc1dd4c4c31f0